eng/docgeneration/docindex.yml

```diff
@@ -0,0 +1,119 @@
+resources:
+  repositories:
+    - repository: azure-sdk-tools
+      type: github
+      name: azure/azure-sdk-tools
+      endpoint: azure
+
+jobs:
+  - job: GenerateDocIndex
+    variables:
+      - template: ../pipelines/templates/variables/globals.yml
+    pool:
+      vmImage: windows-2019
+    steps:
+      - pwsh: |
+          Invoke-WebRequest -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
+          -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
+        workingDirectory: $(Build.BinariesDirectory)
+        displayName: Download and Extract DocFX
+      - task: CmdLine@2
+        displayName: Provision DocFX Directory
+        inputs:
+          script: $(Build.BinariesDirectory)/docfx/docfx.exe init -q
+          workingDirectory: $(Build.SourcesDirectory)
+          failOnStderr: true
+      - pwsh: New-Item -Path . -Name "templates" -ItemType "directory"
+        displayName: Create Template Directory
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/
+      - task: CopyFiles@2
+        displayName: Copy Template to templates folder
+        inputs:
+          sourceFolder: $(Build.SourcesDirectory)/eng/docgeneration/templates
+          content: '**\*'
+          targetFolder: $(Build.SourcesDirectory)/docfx_project/templates
+          overWrite: true
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/templates
+      - pwsh: |
+          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/docfx.json" -Destination "$(Build.SourcesDirectory)/docfx_project/" -Force
+        displayName: Copy over docfx.json
+      - pwsh: |
+          $ServiceList = Get-ChildItem "$(Build.SourcesDirectory)/sdk" -Directory -Exclude eng, mgmtcommon, template | Sort-Object
+          $YmlPath = "$(Build.SourcesDirectory)/docfx_project/api"
+          New-Item -Path $YmlPath -Name "toc.yml" -Force
+          foreach ($Dir in $ServiceList)
+          {
+              New-Item -Path $YmlPath -Name "$($Dir.Name).md" -Force
```

Comment:
@scbedd what do you think about mirroring the index page files after what we have in the main repo? So instead of having a flat folder of html files one-per-service we instead have "sdk\<service>\index.html" and we can even consider having "sdk\<service>\<package>\index.html" as well? That would make file structure and more importantly the links, between the main repo and github.io the same and make generating our release notes with pointers to them much easier to reason about.

Reply:
I think we could have "sdk<service>\index.html" but having "sdk<service><package>\index.html" will mean changing the current layout of the index and putting each package on it  own page. We decided not to do that because each page becomes a scantly white page with little information.